gptel-openai.el

```diff
@@ -151,7 +151,8 @@ Throw an error if there is no match."
                    (:copier gptel--copy-backend))
   name host header protocol stream
   endpoint key models url request-params
-  curl-args)
+  curl-args
+  (coding-system nil :documentation "Can be set to `binary' if the backend expects non UTF-8 output."))
```

Comment:
The documentation for this can be more general.  Perhaps something like:

```
"The coding system used by the backend, defaults to utf-8-unix.  Does not apply to Windows systems."
```

Reply:
Currently we only handle the case where `(eq (gptel-backend-coding-system backend) 'binary)`. If we want to support any coding system, do we want to support separate encoding and decoding systems? Or alternatively we could rename this field to something like `use-binary-coding` to keep the options as utf8 or binary

Reply:
You're right.  Let's leave it at `coding-system` and with your current documentation.  Supporting any coding system is a messy affair because of Windows.